test/components/views/rooms/RoomListHeader-test.tsx

```diff
@@ -28,12 +30,97 @@ import DMRoomMap from "../../../../src/utils/DMRoomMap";
 import MatrixClientContext from "../../../../src/contexts/MatrixClientContext";
 import SettingsStore from "../../../../src/settings/SettingsStore";
 import { SettingLevel } from "../../../../src/settings/SettingLevel";
+import { shouldShowComponent } from '../../../../src/customisations/helpers/UIComponents';
+import { UIComponent } from '../../../../src/settings/UIFeature';
+
+jest.mock('../../../../src/customisations/helpers/UIComponents', () => ({
+    shouldShowComponent: jest.fn(),
+}));
+
+const blockUIComponent = component => {
+    mocked(shouldShowComponent).mockImplementation(feature => feature !== component);
+};
+
+const setupSpace = (client) => {
```

Comment:
Likewise, please add types for `client` and the return value, and do the same for all the other functions you've added

Reply:
done